docs/api/components/schemas.yaml

```
@@ -1719,9 +1719,23 @@ Database:
       default: true
       deprecated: true
     num_index_replicas:
-      description: This is the number of Global Secondary Indexes (GSI) to use for core indexes.
+      description: |-
+        **Deprecated, please use the database setting `index.num_replicas` instead**
+      deprecated: true
       type: number
       default: 1
+    index:
+      type: object
+      description: Settings for Global Secondary Indexes (GSI).
+      properties:
+        num_replicas:
+          description: This is the number of Global Secondary Indexes (GSI) to use for core indexes.
+          type: number
+          default: 1
+        num_partitions:
+          description: The number of partitions to use for the Global Secondary Indexes (GSI) for large indexes. It is not recommended to set this unless you have performance issues. If set, the recommended number is 8.
```

Going to preempt some user-facing documentation around this which calls out performance tradeoffs and any required migration steps, since if this option is available, some users are going to try without completely comprehending.

Feel free to push back or change based on what you know/think we'll be doing around docs.

I used this page/section as a reference for this suggestion:

https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/index-partitioning.html#performance-considerations 

```suggestion
          description: The number of partitions to use for the large indexes created by Sync Gateway. It is not recommended to set this unless you require additional horizontal scalability for individual indexes and have appropriately scaled your Query nodes to handle the increased query parallelism. If set, the recommended number is 8 and does not need to be directly related to the number of your Query nodes. Ensure documentation is read to understand the performance tradeoffs and instructions for migration if you have previously run with only one partition.
```